src/object_based/extent.hpp

```diff
@@ -21,18 +21,44 @@ namespace ear {
    */
   double extentMod(double extent, double distance);
 
+  /// calculates the extent weight for a given angle, in terms of the sin or
+  /// cos of that angle
+  class AngleToWeight {
+   public:
+    AngleToWeight();
+    /// angles < start_angle yield 1.0
+    /// angles > end_angle yield 0.0
+    /// other angles are interpolated
+    AngleToWeight(double start_angle, double end_angle);
+
+    double from_cos(double cos_angle) const;
+    double from_sin(double sin_angle) const;
+
+    // is an angle inside or outside the extent shape?
+    // these are here to allow them to be inlined in HTM
+    bool inside_from_cos(double cos_angle) const {
+      return cos_angle >= cos_start_angle;
+    }
+    bool outside_from_cos(double cos_angle) const {
+      return cos_angle <= cos_end_angle;
+    }
+    bool inside_from_sin(double sin_angle) const {
+      return sin_angle <= sin_start_angle;
+    }
+    bool outside_from_sin(double sin_angle) const {
+      return sin_angle >= sin_end_angle;
+    }
+
+   private:
+    double cos_start_angle, cos_end_angle;
```

Comment:
leading underscore for member vars? not really bothered here as the code is compact enough that it's obvious